app/src/main/java/io/gnosis/safe/notifications/NotificationManager.kt

```diff
@@ -1,35 +1,45 @@
 package io.gnosis.safe.notifications
 
-import android.app.Notification
-import android.app.NotificationChannel
+import android.app.*
 import android.app.NotificationManager
-import android.app.PendingIntent
 import android.content.Context
 import android.graphics.Color
 import android.os.Build
+import androidx.annotation.StringRes
 import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationManagerCompat
 import io.gnosis.data.models.Safe
+import io.gnosis.data.repositories.SafeRepository
 import io.gnosis.data.repositories.TokenRepository
 import io.gnosis.safe.R
 import io.gnosis.safe.notifications.models.PushNotification
 import io.gnosis.safe.ui.StartActivity
 import io.gnosis.safe.utils.BalanceFormatter
 import io.gnosis.safe.utils.convertAmount
 import io.gnosis.safe.utils.formatForTxList
+import kotlinx.coroutines.runBlocking
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
 import pm.gnosis.svalinn.common.PreferencesManager
 import pm.gnosis.svalinn.common.utils.edit
 
 class NotificationManager(
     private val context: Context,
     private val preferencesManager: PreferencesManager,
-    private val balanceFormatter: BalanceFormatter
+    private val balanceFormatter: BalanceFormatter,
+    private val safeRepository: SafeRepository,
+    private val notificationManager: NotificationManagerCompat = NotificationManagerCompat.from(context)
 ) {
 
-    private val notificationManager = NotificationManagerCompat.from(context)
-
     init {
-        createNotificationChannel(CHANNEL_ID)
+        if (Build.VERSION.SDK_INT >= 26) {
+            // For upgrading users. Can be removed in a future release?
```

Comment:
can we check if channel group already exists?

Reply:
Apparently, we can. Will rewrite to check first.